src/SparseEnsembleKalmanInversion.jl

```diff
@@ -0,0 +1,153 @@
+#Sparse Ensemble Kalman Inversion: specific structures and function definitions
+using Convex, SCS
+using SparseArrays
+
+"""
+    SparseInversion <: Process
+
+A sparse ensemble Kalman Inversion process
+"""
+struct SparseInversion{FT <: AbstractFloat, IT <: Int} <: Process
+    ""
+    γ::FT
+    ""
+    threshold_eki::Bool
+    ""
+    threshold_value::FT
+    ""
+    reg::FT
+    ""
+    uc_idx::Vector{IT}
+end
+
+"""
+    sparse_qp(ekp::EnsembleKalmanProcess{FT, IT, SparseInversion{FT,IT}}, v_j::Vector{FT}, cov_vv_inv::Array{FT, 2}, H_u::SparseArrays.SparseMatrixCSC{FT}, H_g::SparseArrays.SparseMatrixCSC{FT}, y_j::Vector{FT}, γ::FT; H_uc::SparseArrays.SparseMatrixCSC{FT} = H_u) where {FT, IT} 
+
+Solving quadratic programming problem with sparsity constraint.
+"""
+function sparse_qp(
+    ekp::EnsembleKalmanProcess{FT, IT, SparseInversion{FT, IT}},
+    v_j::Vector{FT},
+    cov_vv_inv::Array{FT, 2},
+    H_u::SparseArrays.SparseMatrixCSC{FT},
+    H_g::SparseArrays.SparseMatrixCSC{FT},
+    y_j::Vector{FT},
+    γ::FT;
+    H_uc::SparseArrays.SparseMatrixCSC{FT} = H_u,
+) where {FT, IT}
+
+    P = H_g' * inv(ekp.obs_noise_cov) * H_g + cov_vv_inv
```

Comment:
Please avoid inverse computations when solving linear systems, do

`P = H_g' * (ekp.obs_noise_cov \ H_g) + cov_vv_inv`

instead.

Reply:
I have updated the implementations to avoid directly calling matrix inverse.